src/components/Card.js

```diff
@@ -46,7 +57,7 @@ export default class Card {
   }
 
   //like function
-  _handleLikeIcon() {
+  handleLikeIcon() {
```

Comment:
Needs correction: You don't need this method anymore as you should toggle likes state only after the successful server response using the provided flag (not just toggle) and save that flag into the class instance

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```